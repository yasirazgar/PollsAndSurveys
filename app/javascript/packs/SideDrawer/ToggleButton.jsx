import React from 'react';

import './ToggleButton.scss';

const ToggleButton = props => (
  <button className="toggle-button" onClick={props.clickHandler}>
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
    <div className="toggle-button__line" />
  </button>

);

export default ToggleButton;
