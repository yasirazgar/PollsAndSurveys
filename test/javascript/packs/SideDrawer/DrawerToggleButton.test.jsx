import React from 'react'
import { shallow, mount } from 'enzyme'
import ToggleButton from 'packs/SideDrawer/ToggleButton'

describe("ToggleButton", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ToggleButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it("clickHandler gets called on clicking ToggleButton", () => {
    const clickHandler = jest.fn();
    const wrapper = mount(<ToggleButton clickHandler={clickHandler} />)

    wrapper.find('.toggle-button').first().simulate('click');

    expect(clickHandler).toHaveBeenCalledTimes(1)
  });
});
