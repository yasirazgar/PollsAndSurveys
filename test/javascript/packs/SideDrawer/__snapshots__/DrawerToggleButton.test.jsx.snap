// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders correctly 1`] = `
<button
  className="toggle-button"
>
  <div
    className="toggle-button__line"
  />
  <div
    className="toggle-button__line"
  />
  <div
    className="toggle-button__line"
  />
</button>
`;
